checker/src/org/checkerframework/checker/nullness/CollectionToArrayHeuristics.java

```diff
@@ -175,8 +175,7 @@ private boolean isHandledArrayCreation(Tree argument, String receiver) {
     private boolean isNonNullReceiver(MethodInvocationTree tree) {
         // check receiver
         AnnotatedTypeMirror receiver = atypeFactory.getReceiverType(tree);
-        AnnotatedDeclaredType collection = (AnnotatedDeclaredType) AnnotatedTypes.asSuper(processingEnv.getTypeUtils(), atypeFactory, receiver, collectionType);
-        assert collection != null;
+        AnnotatedDeclaredType collection = AnnotatedTypes.asSuper(atypeFactory, receiver, collectionType);
 
         if (collection.getTypeArguments().isEmpty()
```

Comment:
Here and in a few places below there are checks whether type arguments are empty in places where that seems impossible. E.g. here the Collection always has a type argument. When are these checks triggered?
I didn't see in `AnnotatedTypes.asSuper` when it might return a raw type.